Gems/PhysX/Code/Editor/ColliderCapsuleMode.h

```diff
@@ -11,13 +11,15 @@
 #include <Editor/Source/ComponentModes/PhysXSubComponentModeBase.h>
 #include <AzToolsFramework/ComponentModes/BoxViewportEdit.h>
 #include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <AzToolsFramework/ComponentModes/CapsuleViewportEdit.h>
 
 namespace PhysX
 {
     /// Sub component mode for modifying the height and radius on a capsule collider.
     class ColliderCapsuleMode
         : public PhysXSubComponentModeBase
         , private AzFramework::EntityDebugDisplayEventBus::Handler
+        , private AzToolsFramework::CapsuleViewportEdit
```

Comment:
suggestion: Not sure how much of a pain this would be, but having `CapsuleViewportEdit` be a member instead of inheriting privately might make reuse a lot easier (we could move this to a separate change but I reckon that makes the API cleaner and easier to reuse with less duplication)

Reply:
The interface might need to be extracted for things like `GetCapsuleWorldTransform` (or we maybe use callbacks/AZStd::functions 🤔?)

Reply:
I actually started out doing it that way to begin with, but then I ended up with loads of callbacks, and then a setter function for each callback, and it all seemed a bit clunky. Happy to discuss further though.

Reply:
Got you, yes in this situation it's a bit of a tricky one 🤔 I think this is a good first transformation for now, and we might want to revisit it in future 👍